### PR TITLE
Visitor can Register for an Account 

### PR DESCRIPTION
**Pull request by beccaburns**
Register for Account
===============

[PT-Link](https://www.pivotaltracker.com/story/show/169253742)

## PR Contents
- User model 
- FactoryBot for user
- Request specs for user
- Authentication & registration spec files
- Validations for user
- Devise Token Auth gem installed

## What we Learned
- Continued learning on how to write specs for unit testing API

### CODE REVIEW

**Review by yasmineezequiel — APPROVED** [review]

**Review by yasmineezequiel — APPROVED** [review]

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    it { is_expected.to have_db_column :email }
+    it { is_expected.to have_db_column :tokens }
+    it { is_expected.to have_db_column :created_at }
+    it { is_expected.to have_db_column :updated_at }
+  end
+
+  describe 'Validations' do
+    it { is_expected.to validate_presence_of(:email) }
+    it { is_expected.to validate_confirmation_of(:password) }
+    #it { is_expected.to validate_presence_of(:nickname) }
```
Comment (shsteindorsson) [inline spec/models/user_spec.rb]: Can we remove the commented line?

**Review by shsteindorsson — APPROVED** [review]

File: index_users_on_city [hunk 2]
```diff
@@ -0,0 +1,2 @@
+      invoke  active_record
+      create    db/migrate/20191026153252_add_index_column_to_user.rb
```
Comment (faraznaeem) [inline index_users_on_city]: What is this?

File: spec/requests/api/v1/registrations_spec.rb [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+                                     password: 'password',
+                                     password_confirmation: 'password'
+                                  }, headers: headers
+      expect(response_json['errors']['email']).to eq ['has already been taken']
+      expect(response.status).to eq 422
+    end
+
+    it 'an already registered nickname' do
+      expect {
+      FactoryBot.create(:user, nickname: 'becca',
```
Comment (faraznaeem) [inline spec/requests/api/v1/registrations_spec.rb]: Do you need to write `FactoryBot.create`?

**Review by faraznaeem — CHANGES_REQUESTED** [review]

**Review by shsteindorsson — APPROVED** [review]

**Review by lucamarial — APPROVED** [review]

File: spec/support/response_json.rb [hunk 4]
```diff
@@ -4,3 +4,6 @@ def response_json
   end
 end
 
+RSpec.configure do |config|
+  config.include ResponseJSON
+end
```
Comment (DIzosimov) [inline spec/support/response_json.rb:9]: Why does this show here, when in the code it is included in the rails_helper instead of response_json?

**Review by DIzosimov — APPROVED** [review]

**Review by faraznaeem — APPROVED** [review]

**Pull request merged**